static/shared/js/emoji.js

```diff
@@ -208,12 +215,24 @@ export function get_emoji_details_by_name(emoji_name) {
             emoji_info.still_url = emoji_code_info.still_url;
         }
     } else {
-        const codepoint = get_emoji_codepoint(emoji_name);
-        if (codepoint === undefined) {
-            throw new Error("Bad emoji name: " + emoji_name);
+        // deactivation is implied by reaching this line
+        const emoji_code = deactivated_emoji_name_to_code.get(emoji_name);
+        if (emoji_code) {
+            const deactivated_realm_emoji = all_realm_emojis.get(emoji_code);
+            emoji_info.reaction_type = "realm_emoji";
+            emoji_info.emoji_code = deactivated_realm_emoji.id;
+            emoji_info.url = deactivated_realm_emoji.emoji_url;
+            if (deactivated_realm_emoji.still_url) {
+                emoji_info.still_url = deactivated_realm_emoji.still_url;
+            }
+        } else {
+            const codepoint = get_emoji_codepoint(emoji_name);
```

Comment:
I think we want this check, for unicode emoji, to be before the deactivated emoji check.